tests/sles4sap/publiccloud/check_ibsm_embargoed_azure.pm

```diff
@@ -0,0 +1,44 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Deployment steps for qe-sap-deployment
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+use strict;
+use warnings;
+use base 'sles4sap_publiccloud_basetest';
+use publiccloud::utils "validate_repo";
+use testapi;
+
+sub test_flags {
+    return {fatal => 1, publiccloud_multi_module => 1};
+}
+
+sub run() {
+    my ($self, $run_args) = @_;
+    my $instance = $run_args->{my_instance};
+    $self->{network_peering_present} = 1 if ($run_args->{network_peering_present});
+    record_info('CONTEXT LOG', "instance:$instance network_peering_present:$self->{network_peering_present}");
+
+    set_var('MAINT_TEST_REPO', get_var('INCIDENT_REPO')) if get_var('INCIDENT_REPO');
+    my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
+    my @repos = split(/,/, get_var('MAINT_TEST_REPO'));
+    my $count = 0;
+
+    while (defined(my $maintrepo = shift @repos)) {
+        next if $maintrepo =~ /^\s*$/;
+        if (!validate_repo($maintrepo)) {
```

Comment:
This `if` is probably a good place for a comment in the test code to explain what the test is doing, and why there's a `!`

Reply:
Added a short description on top, but I'll add a comment here too to explain the `!`

Reply:
maybe one near to the `die` below too could be good

Reply:
does `validate_repo` return 0 only if the update is embargoed? What will happen to the test if the sync system to the IBSm is broken or something like that?

Reply:
Yes, validate_repo will return 0 only if there's an embargoed update. If there are other problems like wrong repo name etc it fails on spot, but without the message saying that there's an embargoed update.

same goes if the connection to ibsm fails, the first `run_ssh_command` will fail but without the embargo message (it uses `assert_script_run` below the hood)